src/redux/sagas/opponents.js

```diff
@@ -0,0 +1,33 @@
+import { fork, take, cancel } from 'redux-saga/effects';
+import { reduxSagaFirebase } from '../../gateways/firebase';
+
+import {
+  types,
+  syncOpponents,
+} from '../actions/opponents';
+
+const opponentsTransformer = ({ value }) => Object.keys(value).map(key => ({
+  ...value[key],
+  id: key
+}));
+
+function * syncOpponentsSaga () {
+  yield fork(
+    reduxSagaFirebase.database.sync,
+    'opponents',
+    {
+      successActionCreator: syncOpponents,
+      transform: opponentsTransformer,
+    }
+  );
+}
+
+export default function * opponentSaga () {
+  while( yield take(types.OPPONENTS.SUBSCRIBE) ) {
```

Comment:
Not sure I fully understand, but doesn't the `yield` keyword always return a truthy value?

>The yield keyword actually returns an IteratorResult object with two properties ...

Reply:
Yep. The think is it doesn’t necessarily return that value immediately - the loop here only runs when a new event is received.